### PR TITLE
Alonzo initial work

### PR DESCRIPTION
**Pull request by rooooooooob**
[ X ] Plutus components generation (initial generation + clean-up / some refactoring)
[ X ] `ConstrPlutusData` custom impl
[ X ] re-generate existing portions of code that made sense to be codegenned again
[ X ] re-write `TransactionMetadata` -> `AuxilaryData` to handle new alonzo format/plutus
[ X ] re-write `TransactioOutput` (codegen can't handle it)
[ X ] re-write `MoveInstantaneousReward`
[ X ] Updates to existing portions of library

### CODE REVIEW

**Comment by SebastienGllmt** [comment]
https://github.com/input-output-hk/cardano-ledger-specs/pull/2355/files this PR makes changes to the CDDL spec

File: rust/src/tx_builder.rs [hunk 1]
```diff
@@ -455,12 +462,17 @@ impl TransactionBuilder {
...
-                Some(x) => Some(utils::hash_metadata(x)),
+                Some(x) => Some(utils::hash_auxiliary_data(x)),
             },
             validity_start_interval: self.validity_start_interval,
             mint: self.mint.clone(),
+            // TODO: update for use with Alonzo
+            script_data_hash: None,
+            collateral: None,
+            required_signers: None,
+            network_id: None,
```
Comment (vantuz-subhuman) [inline rust/src/tx_builder.rs:475]: @rooooooooob , there must be setters added for these new fields, right?

**Review by yushih — COMMENTED** [review]
I think this line https://github.com/Emurgo/cardano-serialization-lib/blob/1a1561df4104f11f44446e5bb9b5a2415b5a7ecd/rust/src/lib.rs#L743 should be changed to `MoveInstantaneousRewardsCert(MoveInstantaneousReward)`.

File: rust/src/tx_builder.rs (hunk 1 above)
Comment (rooooooooob) [reply rust/src/tx_builder.rs:475]: Maybe, I just updated the bare minimum in the tx builder as those are all optional fields. You can still manually set them by manual building tx. I'm not sure about the tx creation process for all alonzo-era txs so I left it like this.

**Comment by rooooooooob** [comment]
There's been another update regarding some of the bigint types after a discussion with Jared that was made to the CDDL today but I will address that in a separate PR later as I will be providing more usable wrappers around those anyway.

**Review by vantuz-subhuman — APPROVED** [review]

**Pull request merged**